kernel/kernel-api/src/test/scala/io/delta/kernel/internal/stats/FileSizeHistogramSuite.scala

```diff
@@ -269,6 +269,100 @@ class FileSizeHistogramSuite extends AnyFunSuite {
     assert(histogram === reconstructedHistogram.get())
   }
 
+  test("plus should combine histograms correctly") {
+    // Create two histograms
+    val histogram1 = FileSizeHistogram.createDefaultHistogram()
+    val histogram2 = FileSizeHistogram.createDefaultHistogram()
+
+    // Add data to first histogram
+    histogram1.insert(1 * MB)
+    histogram1.insert(1 * MB)
+    histogram1.insert(10 * MB)
+
+    // Add data to second histogram
+    histogram2.insert(2 * MB)
+    histogram2.insert(10 * MB)
+    histogram2.insert(10 * MB)
+
+    // Combine histograms
+    val combined = histogram1.plus(histogram2)
+
+    // Check results
+    val mbBinIndex1 = getBinIndexForTesting(combined.getSortedBinBoundaries, 1 * MB)
+    val mbBinIndex2 = getBinIndexForTesting(combined.getSortedBinBoundaries, 2 * MB)
+    val mbBinIndex10 = getBinIndexForTesting(combined.getSortedBinBoundaries, 10 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex1) == 2)
+    assert(combined.getTotalBytes(mbBinIndex1) == 2 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex2) == 1)
+    assert(combined.getTotalBytes(mbBinIndex2) == 2 * MB)
+
+    assert(combined.getFileCounts(mbBinIndex10) == 3)
+    assert(combined.getTotalBytes(mbBinIndex10) == 30 * MB)
```

Comment:
this works for me, another idea for testing though is to make a copy of histogram1 and then do all the operations to both histogram2 & histogram1_copy --> histogram1_copy = histogram1.plus(histogram2)